pkg/sigo/kdtree.go

```diff
@@ -147,17 +154,50 @@ func (n *node) build() {
 		}
 
 		n.cluster = nil
-		n.subNodes[0].build()
-		n.subNodes[1].build()
+		err = n.subNodes[0].build()
+
+		if err != nil {
+			return err
+		}
+
+		err = n.subNodes[1].build()
+		if err != nil {
+			return err
+		}
 	}
+
+	return nil
 }
 
 // split creates 2 subnodes by ordering the node and splitting in order to have 2 equal parts
 // and all elements having the same value in the same subnode.
-func (n *node) split() (node, node, bool) {
-	sort.SliceStable(n.cluster, func(i int, j int) bool {
-		return n.cluster[i].QuasiIdentifer()[n.rot] < n.cluster[j].QuasiIdentifer()[n.rot]
-	})
+func (n *node) split() (node, node, bool, error) {
+	var globalError error
+
+	less := func(i, j int) bool {
+		valueI, err := n.cluster[i].QuasiIdentifer()
+		if err != nil {
+			// Stocker l'erreur dans la variable globale
```

Comment:
please use English

Reply:
Yes, adding a step Data Validation is a better way to handling error. I will revert to the first commit just keep the venom test.

I propose adding an new interface `dataValidator` in case we use other type than `float64` in the future. After by default we use `float64DataValidator` after source created to valide input data. After we can focus in other step of the workflow.